pkg/api/handlers/libpod/pods.go

```diff
@@ -25,24 +31,75 @@ func PodCreate(w http.ResponseWriter, r *http.Request) {
 		runtime = r.Context().Value("runtime").(*libpod.Runtime)
 		err     error
 	)
-	var psg specgen.PodSpecGenerator
-	if err := json.NewDecoder(r.Body).Decode(&psg); err != nil {
-		utils.Error(w, "failed to decode specgen", http.StatusInternalServerError, errors.Wrap(err, "failed to decode specgen"))
+
+	opts := &entities.PodCreateOptions{Infra: true, NetFlags: &entities.NetFlags{}, Net: &entities.NetOptions{}}
+	buffer := bytes.NewBuffer(make([]byte, 0))
+	r.Body = ioutil.NopCloser(io.TeeReader(r.Body, buffer)) // create teereader so we can decode body twice, once for podOpts, once for Infra
+	if err := json.NewDecoder(r.Body).Decode(opts); err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "failed to decode specgen"))
+		return
+	}
+	flags, err := common.GenerateTempNetFlags(opts.NetFlags, opts.Infra) // a good workaround, allowing us to temprarily make netflags and use the netflags -> netOpts
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "failed to generate net flags"))
 		return
 	}
-	// parse userns so we get the valid default value of userns
-	psg.Userns, err = specgen.ParseUserNamespace(psg.Userns.String())
+
+	opts.Net, err = common.NetFlagsToNetOptions(opts.Net, flags, false, nil) // create net options from the flags
 	if err != nil {
-		utils.Error(w, "failed to parse userns", http.StatusInternalServerError, errors.Wrap(err, "failed to parse userns"))
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "failed to convert net flags to net options"))
 		return
 	}
-	pod, err := generate.MakePod(&psg, runtime)
+	if opts.Net == nil {
+		opts.Net = &entities.NetOptions{}
+	}
+
+	podSpec := specgen.NewPodSpecGenerator()
+	psg, err := entities.ToPodSpecGen(*podSpec, opts)
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "failed to decode specgen"))
+		return
+	}
+	if !psg.NoInfra {
+		infraOptions := &entities.ContainerCLIOpts{ImageVolume: "bind", IsInfra: true, Net: &entities.NetOptions{}}
+		if err := json.NewDecoder(buffer).Decode(infraOptions); err != nil {
+			utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "failed to decode specgen"))
+			return
+		}
+		imageName := psg.InfraImage
+		rawImageName := psg.InfraImage
+		if imageName == "" {
+			imageName = "k8s.gcr.io/pause:3.5"
+			rawImageName = "k8s.gcr.io/pause:3.5"
+		}
+		curr := infraOptions.Quiet
+		infraOptions.Quiet = true
+		name, err := containers.PullImage(imageName, *infraOptions)
```

Comment:
The backend shouldn't call front-end things (i.e., `cmd/podman` should not be called outside of that very folder). Please use `libimage.Pull` directly.